lib/logstash/outputs/sns.rb

```diff
@@ -70,54 +63,45 @@ def register
   def receive(event)
     return unless output?(event)
 
-    arn     = Array(event["sns"]).first || @arn
-
-    raise "An SNS ARN required." unless arn
-
-    message = Array(event["sns_message"]).first
-    subject = Array(event["sns_subject"]).first || event["host"]
-
-    # Ensure message doesn't exceed the maximum size.
-    if message
-      # TODO: Utilize `byteslice` in JRuby 1.7: http://jira.codehaus.org/browse/JRUBY-5547
-      message = message.slice(0, MAX_MESSAGE_SIZE_IN_BYTES)
+    if (sns_msg = Array(event["sns_message"]).first)
+      send_sns_message(event_arn(event), event_subject(event), sns_msg)
     else
-      if @format == "plain"
-        message = self.class.format_message(event)
-      else
-        message = self.class.json_message(event)
-      end
+      @codec.encode(event)
     end
+  end
 
-    # Log event.
-    @logger.debug("Sending event to SNS topic [#{arn}] with subject [#{subject}] and message:")
-    message.split("\n").each { |line| @logger.debug(line) }
-
-    # Publish the message.
-    @sns.topics[arn].publish(message, :subject => subject.slice(0, MAX_SUBJECT_SIZE_IN_CHARACTERS))
+  private
+  def publish_boot_message_arn
+    # Try to publish a "Logstash booted" message to the ARN provided to
+    # cause an error ASAP if the credentials are bad.
+    if @publish_boot_message_arn
+      @sns.topics[@publish_boot_message_arn].publish("Logstash successfully booted", :subject => "Logstash booted")
+    end
   end
 
-  def self.json_message(event)
-    json      = event.to_json
-    json_size = json.bytesize
+  private
+  def send_sns_message(arn, subject, message)
+    raise ArgumentError, "An SNS ARN is required." unless arn
```

Comment:
I agree with the idea here, but I think that would violate the contract of the plugin API. The original author had it publish a boot message to essentially test the connection, so this check will be run anyway.

Additionally, this plugin can direct the output to an ARN specified by a message, so we need to do check on each send anyway.

One could, in fact, setup this plugin with the boot message disabled and explicit ARNs sent with each message (maybe via add_field), in which case a check in register would break such a configuration.